docs/developer/release/3-update-version-in-code.md

```diff
@@ -1,60 +1,74 @@
+  The VERSION file is already pointing to the version being released ever since the `release/` branch was created.
+* [Patch Release](https://github.com/grafana/alloy/pull/1767)
 
-    1. `CHANGELOG.md` Header
-        - First Release Candidate or a Patch Release
-            - Add a new header under `Main (unreleased)` for `VERSION`.
-        - Additional RCV or SRV
-            - Update the header `PREVIOUS_RELEASE_CANDIDATE_VERSION` to `VERSION`. The date may need updating.
+### 1.1. Add the new version to the CHANGELOG
 
-    2. Move the unreleased changes we want to add to the release branch from `Main (unreleased)` to `VERSION`.
+For a First Release Candidate (`rc.0`), replace the `Main (unreleased)` header with a new one for  `VERSION`.
 
-3. Create a PR to merge to main (must be merged before continuing).
+For an Additional Release Candidate or SRV, update the header `PREVIOUS_RELEASE_CANDIDATE_VERSION` to `VERSION`.
 
-    - Release Candidate example PR [here](https://github.com/grafana/agent/pull/3065)
-    - Stable Release example PR [here](https://github.com/grafana/agent/pull/3119)
-    - Patch Release example PR [here](https://github.com/grafana/agent/pull/3191)
+For a patch release, add a new header for `VERSION`.
 
-4. If one doesn't exist yet, create a branch called `release/VERSION_PREFIX` for [grafana/alloy](https://github.com/grafana/alloy).
+### 1.2. Cherry pick commits
 
-5. Cherry pick the commit on main from the merged PR in Step 3 from main into the branch from Step 4:
+If you need certain changes on the release branch but they're not yet there, cherry-pick them onto the release branch.
+In the CHANGELOG, make sure they are listed under the header for the new VERSION and not under `Main (unreleased)`.
 
-    ```
-    git cherry-pick -x COMMIT_SHA
-    ```
+### 1.3. Update the VERSION file
 
-    Delete the `Main (unreleased)` header and anything underneath it as part of the cherry-pick. Alternatively, do it after the cherry-pick is completed.
+The [VERSION file][VERSION-file] on the `release/` branch should point to the stable (or patch) version you're about to release.
 
-6. **If you are creating a patch release,** ensure that the file called `VERSION` in your branch matches the version being published, without any release candidate or build information:
+The contents of the VERSION file should not contain `rc` information.
+Therefore, there is no need to update the VERSION file for additional release candidates (e.g. `rc.1`, `rc.2`).
 
-   > **NOTE**: Only perform this step for patch releases, and make sure that
-   > the change is not pushed to the main branch.
+For example:
+* If you are going to release `v1.2.0-rc.0`, then the VERSION file should contain `v1.2.0`.
+* If you are going to release `v1.5.1`, then the VERSION file should contain `v1.5.1`.
 
-   After updating `VERSION`, run:
+## 2. Update the "main" branch
 
-   ```bash
-   make generate-versioned-files
-   ```
```

Comment:
I think that we should keep this info

Reply:
True, I must have removed it by mistake. Thanks for spotting it! I added it back.